addons/mail/models/discuss/res_partner.py

```diff
@@ -91,4 +91,12 @@ def get_mention_suggestions_from_channel(self, channel_id, search, limit=8):
             },
         )
         store.add(partners)
+        store.add({
+            'partnersWithAccess': [
+                ("ADD", [
+                    {"id": p.id, "type": "partner"}
+                    for p in partners.filtered(lambda partner: not partner.partner_share)
+                ])
+            ],
+        })
```

Comment:
`partnersWithAccess` is a relational field between a thread and a partner. Putting on `store` is too broad, leading to making it too dynamic and limited to a single use on UI.
Also you couple fetch and search together which is wrong: search may happen without fetch, so you could suggest wrong partners in another channel by mistake.

Instead of code with such imperative code, make the code more declarative to the modelling at hand.
`partnersWithAccess` should be a relational field on `Thread`. That way, there's no mandatary need to clear all the time for correctness (clearance is optimisation, no change of correctness

I'll put more precise comment in other parts of code. Here the `partnerWithAccess` should be added in store as a relational field on the given thread.